src/QirRuntime/test/FullstateSimulator/qir-test-simulator.ll

```diff
@@ -0,0 +1,1092 @@
+

```

Comment:
Is this file an expected output to be compared to the actual output generated from the `qir-test-simulator.qs`?
If yes then I fail to understand in which fragments this expected output file is compared to the actual output file. Please advise.

Reply:
Right now the foo.ll files in this project are generated manually from the corresponding foo.qs files. There is no validation that the QIR itself is correct or matches a particular expected pattern (that probably should be done as part of compiler tests for QIR generation). The purpose of these tests is to check that runtime handles the QIR correctly. For example, the tests in qir-static use various mock simulators to validate expectations. Unfortunately, the fullstate simulator would produce random outcomes for most sensible operations. I suppose, I can update the test to handle some situations that guarantee non-random outcomes (like ensuring that R and Adjoint R do cancel each other). Do you think that would be useful?

Reply:
Looks like I still need to get a better understanding of the terminology and mechanisms of the build-and-test process, in order to fully understand your answer. That is why I don't want this my topic to be treated as an obstacle for moving forward with completing this PR. Feel free to go ahead with the PR, no need to answer to the text below.

Let me summarize what I understand so far.
Some `.ll` files are written manually (e.g. `qsharp-runtime\src\QirRuntime\lib\QIR\bridge-*.ll`. They don't have the corresponding `.qs` files). 
But some `.ll` files can be generated from the corresponding `.qs` files. For such `.ll` files we don't need to store their commit history in Git, we can generate them upon need with the build scripts (CMake files). But there are reasons why it is better to generate them manually from time to time and store them in the Git history.

Reply:
Ah! I see what you mean. Yes, in the future we want to be able to generate the test *.ll files from the corresponding *.qs files as part of the build. There are a few obstacles to that at the moment:
- QIR generation pipeline isn't multi-platform yet (Stefan is working on it)
- Some minor manual edits are needed to the generated QIR to make it compile, some of these are because of bugs that Bettina is looking into and some are caused by using multiple *.ll files in a single project -- we might need to restructure the tests because of that.

The bottom line is, that _for now_ the test *.ll files are being generated manually, doctored a bit and then checked in.

Reply:
Logged https://github.com/microsoft/qsharp-runtime/issues/498 to track this